src/api/routes/ws/ws-rpc.ts

```diff
@@ -68,6 +80,24 @@ class SubscriptionManager {
         this.subscriptions.delete(topicId);
       }
     }
+    this.liveSockets.delete(client);
+  }
+
+  setUpHeartbeat() {
+    setInterval(() => {
```

Comment:
`setInterval` returns a timer object https://nodejs.org/api/timers.html#class-timeout -- does this need to be cleared during server shutdown?